### ISSUE TITLE
Lint warning Avoid using dynamic typ on extension type with version 1.1.4

### ISSUE THREAD

**Issue by quoc-huynh-cosee** [bancolombia/dart-code-linter#142]
**Describe the error**
With the version **1.1.4** the **dart_code_linter** package, the following code raises a lint warning on the line `final Asset asset`:

```dart
extension type Asset(String path) { }

class TestClass {
  const TestClass({required this.asset});

  // Lint warning: Avoid using dynamic typ
  final Asset asset;
}
```

This warning is should not be triggered, as the **Asset** is a valid **extension type**. The warning does not occur in version **1.1.3** of **dart_code_linter**

**How ​​to replicate the bug**
Step by step to reproduce the error:
1. Use dart_code_linter version 1.1.4.
2. Define an extension type and use it as a class property as shown in the code above.

**Comment by santitigaga** [comment]
@quoc-huynh-cosee  You are right, It turns out that this functionality was released in a stable form in Dart version 3.3.0 and that scenario was not contemplated.

I hope to fix it tomorrow,Thanks for reporting it

**Comment by santitigaga** [comment]
@quoc-huynh-cosee This will be deployed in the next release
